invisible_cities/cities/components.py

```diff
@@ -601,3 +675,52 @@ def waveform_integrator(limits):
     def integrate_wfs(wfs):
         return cf.spaced_integrals(wfs, limits)[:, ::2]
     return integrate_wfs
+
+
+# Compound components
+def compute_and_write_pmaps(detector_db, run_number, pmt_samp_wid, sipm_samp_wid,
+                  s1_lmax, s1_lmin, s1_rebin_stride, s1_stride, s1_tmax, s1_tmin,
+                  s2_lmax, s2_lmin, s2_rebin_stride, s2_stride, s2_tmax, s2_tmin, thr_sipm_s2,
+                  h5out, compression, sipm_rwf_to_cal=None):
+
+    # Filter events without signal over threshold
+    indices_pass    = fl.map(check_nonempty_indices,
+                             args = ("s1_indices", "s2_indices"),
+                             out = "indices_pass")
+    empty_indices   = fl.count_filter(bool, args = "indices_pass")
+
+    # Build the PMap
+    compute_pmap     = fl.map(build_pmap(detector_db, run_number, pmt_samp_wid, sipm_samp_wid,
+                                         s1_lmax, s1_lmin, s1_rebin_stride, s1_stride, s1_tmax, s1_tmin,
+                                         s2_lmax, s2_lmin, s2_rebin_stride, s2_stride, s2_tmax, s2_tmin, thr_sipm_s2),
+                              args = ("ccwfs", "s1_indices", "s2_indices", "sipm"),
+                              out  = "pmap")
+
+    # Filter events with zero peaks
+    pmaps_pass      = fl.map(check_empty_pmap, args = "pmap", out = "pmaps_pass")
+    empty_pmaps     = fl.count_filter(bool, args = "pmaps_pass")
+
+    # Define writers...
+    write_pmap_         = pmap_writer         (h5out, compression=compression)
+    write_indx_filter_  = event_filter_writer (h5out, "s12_indices", compression=compression)
+    write_pmap_filter_  = event_filter_writer (h5out, "empty_pmap" , compression=compression)
+
```

Comment:
The first bracket doesn't seem to align with anything in particular so I'd move them back a space so that they're not floating

Reply:
You were right, now realigned.